src/components/MovieDetails/MovieDetails.jsx

```diff
@@ -1,37 +1,66 @@
-import React from 'react';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
+import fetchMovie from '../../actions/singleMovie';
 import styles from './MovieDetails.scss';
 
-const MovieDetails = () => (
-  <div className={styles.movieDetails}>
-    <div className={styles.backBtn}>
-      <Link to="/" className={styles.btn}>Back to Search</Link>
-    </div>
-    <div className={styles.movieAvatar}>
-      <img
-        src="http://media.comicbook.com/2017/06/the-batman-movie-fan-poster-ben-affleck-1000887.png"
-        alt="Batman movie poster"
-      />
-    </div>
-    <div className={styles.movieDescription}>
-      <div className={styles.title}>
-        The Batman <span className={styles.rated}>9.9</span>
+class MovieDetails extends Component {
+  componentDidMount() {
+    const { match, dispatch } = this.props;
+    const id = match.params.id;
+    dispatch(fetchMovie(id));
+  }
+
+  componentWillReceiveProps({ location }) {
+    const { dispatch } = this.props;
+    if (/film/.test(location.pathname)) {
+      if (location.pathname === this.props.location.pathname) {
+        return;
+      }
+      const nextId = location.pathname.split('/')[2];
```

Comment:
We can use match object of the router to get params: https://reacttraining.com/react-router/web/example/url-params